code/game/objects/items/plushes.dm

```diff
@@ -407,108 +411,320 @@
+	if(fighting)
+		to_chat(user, "<span class='warning'>[src] is busy fighting!</span>")
+		return FALSE
+	return ..()
+
+/obj/item/toy/plush/fighter/Moved(atom/OldLoc, Dir)
+	. = ..()
+	if(fighting)
+		return
+	for(var/obj/item/toy/plush/fighter/Fighter in view(1, src))
+		if(Fighter.plush_type == plush_type)
+			continue
+		if(plush_type != PLUSH_ABNORMALITY)
+			Fighter.Fight()
+		else
+			Fight()
+		break
+	return
+
+/// The target acquisition that starts a fight.
+/obj/item/toy/plush/fighter/proc/Fight()
+	fighting = TRUE
+	say(pick(fight_start_lines))
+	sleep(1 SECONDS)
+	threats = list()
+	for(var/obj/item/toy/plush/fighter/F in view(7, src))
+		if(F.plush_type == plush_type) // Abnos don't fight abnos, Sinners don't fight Sinners, Magical Girls don't fight Magical Girls.
+			continue
+		threats |= F
+		F.fighting = TRUE
+		F.say(pick(F.fight_start_lines))
+	if(threats.len < 1)
+		fighting = FALSE
+		return
+	Conflict()
+
+/// The fight loop and decision.
+/obj/item/toy/plush/fighter/proc/Conflict()
+	var/atom/throw_target
+	var/loop_count = 0
+	do
+		for(var/obj/item/toy/plush/fighter/dante/D in threats)
+			for(var/obj/item/toy/plush/fighter/Sinner in view(7, D))
+				if(!Sinner.fighting && Sinner.plush_type == PLUSH_SINNER)
+					Sinner.fighting = TRUE
+					threats |= Sinner
+			break
+		var/obj/item/toy/plush/fighter/F = pick(threats)
+		if(!istype(F))
+			threats -= F
+			continue
+		if(istype(F, /obj/item/toy/plush/fighter/dante))
+			continue
+		if(!(F in view(9, src)))
+			F.say("No!")
+			F.visible_message("<span class='warning'>[F] yells as it falls out of the fight!</span>")
+			threats -= F
+			continue
+		if(threats.len <= 0)
+			say("I don’t know what awaits me ahead. Could it be a cliff, or a ditch?")
+			fighting = FALSE
+			return
+		if(F.plush_type == PLUSH_SINNER && prob(20))
+			F.visible_message("[F] is killed by [src]!")
+			threats -= F
+			F.FightLose()
+		else
+			if(prob(80))
+				throw_target = pick(get_adjacent_open_turfs(src))
+				F.throw_at(throw_target, 4, 2, null, TRUE)
+			if(prob(10))
+				F.say(pick(F.in_combat_lines))
+			sleep(25/(threats.len+loop_count))
+			if(F.fight_sfx.len <= 0)
+				playsound(F, pick(generic_sfx), rand(40, 60))
+			else
+				playsound(F, pick(F.fight_sfx), rand(40, 60))
+		var/list/targets = threats.Copy()
+		targets -= F
+		if(targets.len <= 0)
+			targets += F
+		throw_target = pick(get_adjacent_open_turfs(pick(targets)))
+		throw_at(throw_target, 5, 3, null, TRUE)
+		if(prob(5))
+			say(pick(in_combat_lines))
+		if(fight_sfx.len <= 0)
+			playsound(src, pick(generic_sfx), rand(40,60))
+		else
+			playsound(src, pick(fight_sfx), rand(40, 60))
+		loop_count++
+		sleep(30/(threats.len+loop_count))
+	while(prob(100 + threats.len - loop_count))
+	if(prob(CalculateWin()))
+		return FightWin()
+	return FightLose()
+
+/// Does the math for if you win or not, returns a % win chance.
+/obj/item/toy/plush/fighter/proc/CalculateWin()
+	fighting_force = initial(fighting_force)
```

Comment:
Var shadowing like this is really bad. Replace it with something like `effective_force`, and also I don't think you need to be using `initial()` there at all.